modules/json/filterx-cache-json-file.c

```diff
@@ -53,13 +57,28 @@ cache_json_file_error_quark(void)
   return g_quark_from_static_string("filterx-function-cache-json-file-error-quark");
 }
 
+#define FROZEN_OBJECTS_HISTORY_SIZE 5
+
 typedef struct FilterXFunctionCacheJsonFile_
 {
   FilterXFunction super;
   gchar *filepath;
-  FilterXObject *cached_json;
+  gpointer cached_json;
   GPtrArray *frozen_objects;
-} FilterXFuntionCacheJsonFile;
+  GPtrArray *frozen_objects_history[FROZEN_OBJECTS_HISTORY_SIZE];
+  gint history_index;
+  FileMonitor *file_monitor;
+} FilterXFunctionCacheJsonFile;
+
+static void
+_archive_frozen_objects(FilterXFunctionCacheJsonFile *self)
+{
+  if (!self->frozen_objects)
+    return;
+  self->frozen_objects_history[self->history_index] = self->frozen_objects;
+  self->history_index++;
```

Comment:
Can we add an assert not to overflow the buffer in case the cleanup mechanism fails or someone refactors the guarantees away?